src/viam/ur/module/ur_arm_state_controlled.cpp

```diff
@@ -62,56 +62,111 @@ std::optional<URArm::state_::event_variant_> URArm::state_::state_controlled_::h
-                urcl::control::TrajectoryControlMessage::TRAJECTORY_START, static_cast<int>(num_samples), RobotReceiveTimeout::off())) {
-            VIAM_SDK_LOG(error) << "send_trajectory driver->writeTrajectoryControlMessage returned false; dropping connection";
-            std::exchange(state.move_request_, {})->complete_error("failed to send trajectory start message to arm");
-            return event_connection_lost_::trajectory_control_failure();
-        }
-
-        VIAM_SDK_LOG(info) << "URArm::send_trajectory sending " << num_samples << " cubic writeTrajectorySplinePoint";
-        for (size_t i = 0; i < num_samples; i++) {
-            if (!arm_conn_->driver->writeTrajectorySplinePoint(samples[i].p, samples[i].v, samples[i].timestep)) {
-                VIAM_SDK_LOG(error) << "send_trajectory cubic driver->writeTrajectorySplinePoint returned false; dropping connection";
-                std::exchange(state.move_request_, {})->complete_error("failed to send trajectory spline point to arm");
-                return event_connection_lost_::trajectory_control_failure();
+    auto& mr = *state.move_request_;
+
+    return std::visit(
+        [this, &state, &mr](auto& cmd) -> std::optional<event_variant_> {
+            using T = std::decay_t<decltype(cmd)>;
+
+            if constexpr (std::is_same_v<T, std::vector<trajectory_sample_point>>) {
+                // Operating in joint-space
+
+                if (!cmd.empty() && !mr.cancellation_request) {
+                    // Have samples to send, no cancellation requested
+                    VIAM_SDK_LOG(info) << "URArm sending trajectory";
+
+                    auto to_send = std::move(cmd);
+                    const auto num_samples = to_send.size();
+
+                    if (!arm_conn_->driver->writeTrajectoryControlMessage(urcl::control::TrajectoryControlMessage::TRAJECTORY_START,
+                                                                          static_cast<int>(num_samples),
+                                                                          RobotReceiveTimeout::off())) {
+                        VIAM_SDK_LOG(error) << "send_trajectory: start failed; dropping connection";
+                        std::exchange(state.move_request_, {})->complete_error("failed to send trajectory start");
+                        return event_connection_lost_::trajectory_control_failure();
+                    }
+
+                    for (size_t i = 0; i < num_samples; ++i) {
+                        if (!arm_conn_->driver->writeTrajectorySplinePoint(to_send[i].p, to_send[i].v, to_send[i].timestep)) {
+                            VIAM_SDK_LOG(error) << "send_trajectory: spline point failed; dropping connection";
+                            std::exchange(state.move_request_, {})->complete_error("failed to send trajectory spline point");
+                            return event_connection_lost_::trajectory_control_failure();
+                        }
+                    }
+
```

Comment:
prob need to format this one
```suggestion
                    VIAM_SDK_LOG(debug) << "URArm trajectory sent";
```